src/content/en/fundamentals/instant-and-offline/service-worker-lifecycle/index.md

```diff
@@ -0,0 +1,678 @@
+
+To get the update, close or navigate-away all tabs using the current service worker. Then, when you [navigate to the demo again](https://cdn.rawgit.com/jakearchibald/80368b84ac1ae8e229fc90b3fe826301/raw/ad55049bee9b11d47f1f7d19a73bf3306d156f43/index-v2.html){: .external}, you should see the horse. 
+
+This pattern is similar to how Chrome updates. Updates to Chrome download in the background, but don't apply until Chrome restarts. In the mean time, you can continue to use the current version without disruption. However, this is a pain during development, but devtools has ways to make it easier, which I'll cover later in this article.
+
+### Activate
+
+This fires once the old service worker is gone, and your new service worker is able to control clients. This is the ideal time to do stuff that you couldn't do while the old worker was still in use, such as migrating databases and clearing caches.
+
+In the demo above, I maintain a list of caches that I expect to be there, and in the "activate" event I get rid of any others, which will remove the old `static-v1` cache.
+
+Bear in mind you may not be updating from the previous version, it may be a service worker many versions old.
+
+If you pass a promise to `event.waitUntil()` it'll buffer functional events ("fetch", "push", "sync" etc) until the promise resolves, so when your "fetch" event fires, the activation is fully complete.
+
+Caution: The cache storage API is "origin storage" (like localStorage, and IndexedDB). If you run many sites on the same origin (eg `yourname.github.io/myapp`), careful you don't delete caches for your other sites! To avoid this, give your cache names a prefix unique to the current site, eg `myapp-static-v1`, and don't touch caches unless they begin `myapp-`.
+
+### Skip the waiting phase
+
+The waiting phase means you're only running one version of your site at once, but if you don't need that feature, you can make your new service worker activate sooner by calling `self.skipWaiting()`.
+
+This causes your service worker to kick out the current active worker & activate itself once it enters the waiting phase (or immediately if it's already in the waiting phase). It *doesn't* cause your worker to skip installing, just waiting.
+
+It doesn't really matter when you call `skipWaiting`, as long as it's during or before waiting. It's pretty common to call it in the "install" event:
+
+```js
+self.addEventListener('install', event => {
+  self.skipWaiting();
+
+  event.waitUntil(
+    // caching etc
+  );
+});
+```
+
+…but there are other times you might want to call it, which I cover later in the post.
```

Comment:
I was trying to discourage readers jumping around in the post. But meh I'll add it.